# Author: Jianghao Zhu

In [1]:
from __future__ import print_function
import os
import json
import sys
import numpy as np
import pandas as pd
import pylab as pl
import matplotlib as plt
import scipy
import scipy.stats as st

# !!!!!!!!TODO FOR YOU


## what if we used the values for where the former inmate was or was not "Convicted of a felony" (row 10) in the Recidivism (Years 1-3)?

## Null hypothesis:
### The percentage of former inmates got convicted after 3 years is same or higher than the control group, where significance level $\alpha = 0.05$.
### $H_0: P_0 - P_1 \geq 0$
### $H_a: P_0 - P_1 \leq 0$

### $P_0 = 0.117, P_1= 0.1$

In [2]:
alpha = 0.05

In [3]:
P_0 = 11.7 * 0.01
P_1= 10 * 0.01

In [4]:
n_0 = 409
n_1 = 568

Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

In [5]:
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.107116683726


In [6]:
p = lambda p0, p1, n0, n1: (p0 * n0 + p1 * n1) / (n0 + n1)

se = lambda p, n0, n1: np.sqrt(p * (1 - p) * (1.0 / n0 + 1.0 / n1))

In [7]:
zscore = lambda p0, p1, s : (p0 - p1) / s
z_3y = zscore(P_0, P_1, se(p(P_0, P_1, n_0, n_1), n_0, n_1))
print (z_3y)

0.84764020522


In [8]:
p_3y = st.norm.sf(abs(z_3y))
print (p_3y)

0.198319188134


In [9]:
def report_result(p,a):
    print ('is the p value ' + 
           '{0:.2f} smaller than the critical value {1:.2f}?'.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format(\
                            'rejected' if p < a  else 'not rejected') )

    
report_result(p_3y, alpha)

is the p value 0.20 smaller than the critical value 0.05?
NO!
the Null hypothesis is not rejected


# !!!!!! TODO FOR YOU!
# lets see what the chi sq statistics says about the conviction for fellonies (row 10)

## fill in the contingency table below

|convicted of a felony     |     yes      | no        |                   
|---------------------------|-------------|-----------|----------------|
| test sample               | $0.10*568$  |  $0.90*568$          |    568            |
| control sample            | $0.117*409$ |  $0.883*409$       |    409            |
|                           |           |           |                |
| total                     |  104.653         |          872.347 |         997       |

In [10]:
Ntot = 997

In [11]:
sample_values = np.array([[0.10 * 568, 0.90 * 568], [0.117 * 409, 0.883 * 409]])

In [12]:
from evalChisq import evalChisq

In [13]:
evalChisq(sample_values)

0.71849391750528868

### The chi sq statistics is 0.718 which is less than 3.84 when DOF = 1 and $\alpha = 0.05$, so we can not reject the Null Hypothesis.